src/freedv_api.c

```diff
@@ -381,8 +382,14 @@ void freedv_comptx(struct freedv *f, COMP mod_out[], short speech_in[]) {
 #endif
     
     /* 2400 A and B are handled by the real-mode TX */
-    if(FDV_MODE_ACTIVE( FREEDV_MODE_2400A, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_2400B, f->mode)){
-        codec2_encode(f->codec2, f->tx_payload_bits, speech_in);
+    if ((FDV_MODE_ACTIVE( FREEDV_MODE_2400A, f->mode)) || (FDV_MODE_ACTIVE( FREEDV_MODE_2400B, f->mode)) || (FDV_MODE_ACTIVE( FREEDV_MODE_800XA, f->mode))){
+        /* 800XA has two codec frames per modem frame */
+        if(FDV_MODE_ACTIVE( FREEDV_MODE_800XA, f->mode)){
+            codec2_encode(f->codec2, &f->tx_payload_bits[0], &speech_in[  0]);
+            codec2_encode(f->codec2, &f->tx_payload_bits[4], &speech_in[320]);
+        }else{
+            codec2_encode(f->codec2, f->tx_payload_bits, speech_in);
+        }
```

Comment:
Couple of concerns here:
1. We need to test voice modes carefully, e.g. actually run them manually and listen to the output audio.  Testing voice voice is a bit hard to fully automate :thinking: The ctests get us most of the way, e.g. counting frames gives the demod a good work out.
1. I don't think 2400B can operate with  complex modulator or demodulator samples, it operates in the baseband domain of analog FM modulators/demodulators by design.  I probably need to write a README_freedv.md to talk about this sort of detail,and  describe the waveforms more fully.